# Temel DataFrame Analizi 

`Analiz yaparken aşağıdaki metotları kullanacağız`

* `sum`
* `min`
* `max`
* `median`
* `mean`
* `count`
* `describe`

In [1]:
# Pandas kütüphanesini import edelim
import pandas as pd

In [2]:
# King County House veri setini import edelim
houses = pd.read_csv("../data/kc_house_data.csv")

In [3]:
# Dataframe'i inceleyelim
houses

id             date     price  bedrooms  bathrooms  \
0      7129300520  20141013T000000  221900.0         3       1.00   
1      6414100192  20141209T000000  538000.0         3       2.25   
2      5631500400  20150225T000000  180000.0         2       1.00   
3      2487200875  20141209T000000  604000.0         4       3.00   
4      1954400510  20150218T000000  510000.0         3       2.00   
...           ...              ...       ...       ...        ...   
21608   263000018  20140521T000000  360000.0         3       2.50   
21609  6600060120  20150223T000000  400000.0         4       2.50   
21610  1523300141  20140623T000000  402101.0         2       0.75   
21611   291310100  20150116T000000  400000.0         3       2.50   
21612  1523300157  20141015T000000  325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
0             1180      5650     1.0           0     0  ...      7   
1             2570      7242     2.0           0     0  ...      7   
2              770     10000     1.0           0     0  ...      6   
3             1960      5000     1.0           0     0  ...      7   
4             1680      8080     1.0           0     0  ...      8   
...            ...       ...     ...         ...   ...  ...    ...   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
0            1180              0      1955             0    98178  47.5112   
1            2170            400      1951          1991    98125  47.7210   
2             770              0      1933             0    98028  47.7379   
3            1050            910      1965             0    98136  47.5208   
4            1680              0      1987             0    98074  47.6168   
...           ...            ...       ...           ...      ...      ...   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
0     -122.257           1340        5650  
1     -122.319           1690        7639  
2     -122.233           2720        8062  
3     -122.393           1360        5000  
4     -122.045           1800        7503  
...        ...            ...         ...  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[21613 rows x 21 columns]

In [4]:
# min() metodu, dataframe'e ait olan bütün sütunların(feature'ların) minimum değerini gösterir.
houses.min()

id                       1000102
date             20140502T000000
price                    75000.0
bedrooms                       0
bathrooms                    0.0
sqft_living                  290
sqft_lot                     520
floors                       1.0
waterfront                     0
view                           0
condition                      1
grade                          1
sqft_above                   290
sqft_basement                  0
yr_built                    1900
yr_renovated                   0
zipcode                    98001
lat                      47.1559
long                    -122.519
sqft_living15                399
sqft_lot15                   651
dtype: object

In [5]:
# max() metodu, dataframe'e ait olan bütün sütunların(feature'ların) maksimum değerini gösterir.
houses.max()

id                    9900000190
date             20150527T000000
price                  7700000.0
bedrooms                      33
bathrooms                    8.0
sqft_living                13540
sqft_lot                 1651359
floors                       3.5
waterfront                     1
view                           4
condition                      5
grade                         13
sqft_above                  9410
sqft_basement               4820
yr_built                    2015
yr_renovated                2015
zipcode                    98199
lat                      47.7776
long                    -121.315
sqft_living15               6210
sqft_lot15                871200
dtype: object

In [6]:
# Peki max() ve min() metotlarının sonucunda elde ettiğimiz yapının veri tipi nedir?
type(houses.max())

pandas.core.series.Series

***
**Görüldüğü üzere geriye dönen sonuç, Pandas kütüphanesi ait bir veri tipi olan `Series'tir`.**
***

In [7]:
# sum(), metodu, data frame'in sütunlarında bulunan bütün değerleri toplar. Aşağıdaki sonucu incelediğinizde-
# bazı sütunlarda birden fazla tipte değer olduğu için, sum() metodunun sonucunun biraz tuhaf olduğunu- 
# görebilirsiniz. sum() metodu non-numeric değerlerde düzgün çalışmaz.
houses.sum()

id                                                  98994056770455
date             20141013T00000020141209T00000020150225T0000002...
price                                                11672925008.0
bedrooms                                                     72854
bathrooms                                                 45706.25
sqft_living                                               44952873
sqft_lot                                                 326506890
floors                                                     32296.5
waterfront                                                     163
view                                                          5064
condition                                                    73688
grade                                                       165488
sqft_above                                                38652488
sqft_basement                                              6300385
yr_built                                                  4259

In [8]:
# Yukarıdaki sorunu halletmek için sum() metodunun numeric_only parametresini kullanabiliriz.
# Böylece, yalnızca numerik değerlere sahip sütunlarda sum() metodu çalıştırılır.
houses.sum(numeric_only=True)

id               9.899406e+13
price            1.167293e+10
bedrooms         7.285400e+04
bathrooms        4.570625e+04
sqft_living      4.495287e+07
sqft_lot         3.265069e+08
floors           3.229650e+04
waterfront       1.630000e+02
view             5.064000e+03
condition        7.368800e+04
grade            1.654880e+05
sqft_above       3.865249e+07
sqft_basement    6.300385e+06
yr_built         4.259933e+07
yr_renovated     1.824186e+06
zipcode          2.119759e+09
lat              1.027915e+06
long            -2.641409e+06
sqft_living15    4.293536e+07
sqft_lot15       2.759646e+08
dtype: float64

In [9]:
# Scientific notation
9.899406e+13

98994060000000.0

In [10]:
# Titanic veri setini import edelim.
titanic = pd.read_csv("../data/titanic.csv")

In [11]:
# Titanic veri setinin ilk beş örneğini inceleyelim.
titanic.head()

pclass  survived                                             name     sex  \
0       1         1                    Allen, Miss. Elisabeth Walton  female   
1       1         1                   Allison, Master. Hudson Trevor    male   
2       1         0                     Allison, Miss. Helen Loraine  female   
3       1         0             Allison, Mr. Hudson Joshua Creighton    male   
4       1         0  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female   

      age  sibsp  parch  ticket      fare    cabin embarked boat body  \
0      29      0      0   24160  211.3375       B5        S    2    ?   
1  0.9167      1      2  113781    151.55  C22 C26        S   11    ?   
2       2      1      2  113781    151.55  C22 C26        S    ?    ?   
3      30      1      2  113781    151.55  C22 C26        S    ?  135   
4      25      1      2  113781    151.55  C22 C26        S    ?    ?   

                         home.dest  
0                     St Louis, MO  
1  Montreal, PQ / Chesterville, ON  
2  Montreal, PQ / Chesterville, ON  
3  Montreal, PQ / Chesterville, ON  
4  Montreal, PQ / Chesterville, ON

In [12]:
# Bahsettiğimiz gibi sum() metodu non-numeric değerler üzerinde çok verimli değildir.
# Yukarıda titanic veri setinin ilk beş satırını görüyoruz. sum() metodunu bu veri setinde kullanalım.
titanic.sum()

pclass                                                    3004
survived                                                   500
name         Allen, Miss. Elisabeth WaltonAllison, Master. ...
sex          femalemalefemalemalefemalemalefemalemalefemale...
age          290.91672302548633953714718242680?245032363747...
sibsp                                                      653
parch                                                      504
ticket       2416011378111378111378111378119952135021120501...
fare         211.3375151.55151.55151.55151.5526.5577.958305...
cabin        B5C22 C26C22 C26C22 C26C22 C26E12D7A36C101?C62...
embarked     SSSSSSSSSCCCCSSSCCCCSSCCSCCCSSSCSSSCSSSCCCSCCS...
boat         211???310?D??496B??68A55548?778D?788?469???6D8...
body         ???135?????22124??????????????148?????????????...
home.dest    St Louis, MOMontreal, PQ / Chesterville, ONMon...
dtype: object

In [13]:
# titanic dataframe'inin şekline bakalım
titanic.shape

(1309, 14)

Yukardaki çalışmalara bakarak, 500 kişinin Titanik kazasından kurtulduğunu görebiliyoruz. `titanic.shape` ifadesini kullanarak Titanik gemisinde kaç kişi olduğunu öğrenebiliriz. Artık 1309 kişiden 500'ünün kurtuluduğunu biliyoruz.


In [14]:
# titanic dataframe'i içerisindeki sayısal değerlerin toplamın görelim
titanic.sum(numeric_only=True)

pclass      3004
survived     500
sibsp        653
parch        504
dtype: int64

In [15]:
# nst-est2020 datasetini inceleyelim
pd.read_csv("../data/nst-est2020.csv").head()

SUMLEV REGION DIVISION  STATE              NAME  CENSUS2010POP  \
0      10      0        0      0     United States      308745538   
1      20      1        0      0  Northeast Region       55317240   
2      20      2        0      0    Midwest Region       66927001   
3      20      3        0      0      South Region      114555744   
4      20      4        0      0       West Region       71945553   

   ESTIMATESBASE2010  POPESTIMATE2010  POPESTIMATE2011  POPESTIMATE2012  \
0          308758105        309327143        311583481        313877662   
1           55318414         55380764         55608318         55782661   
2           66929737         66975328         67164092         67348275   
3          114563042        114869421        116019483        117264196   
4           71946912         72101630         72791588         73482530   

   POPESTIMATE2013  POPESTIMATE2014  POPESTIMATE2015  POPESTIMATE2016  \
0        316059947        318386329        320738994        323071755   
1         55912775         56021339         56052790         56063777   
2         67576524         67765576         67885682         68018175   
3        118397213        119666248        121049223        122419547   
4         74173435         74933166         75751299         76570256   

   POPESTIMATE2017  POPESTIMATE2018  POPESTIMATE2019  POPESTIMATE042020  \
0        325122128        326838199        328329953          329398742   
1         56083383         56084543         56002934           55924275   
2         68160342         68263019         68340091           68357895   
3        123611036        124649156        125686544          126494232   
4         77267367         77841481         78300384           78622340   

   POPESTIMATE2020  
0        329484123  
1         55849869  
2         68316744  
3        126662754  
4         78654756

In [16]:
# nst-est2020 dataset'ini, bütün sütunlarının isimlerini küçük harfe dönüştürecek şekilde import edelim.

names = ['sumlev', 'region', 'division', 'state', 'name', 'census2010pop', 'estimatesbase2010', 
         'popestimate2010', 'popestimate2011', 'popestimate2012', 'popestimate2013', 'popestimate2014', 
         'popestimate2015', 'popestimate2016', 'popestimate2017', 'popestimate2018', 'popestimate2019', 
         'popestimate042020', 'popestimate2020']

state_pops = pd.read_csv("../data/nst-est2020.csv", names=names, header=0)


In [17]:
state_pops

sumlev region division  state                  name  census2010pop  \
0       10      0        0      0         United States      308745538   
1       20      1        0      0      Northeast Region       55317240   
2       20      2        0      0        Midwest Region       66927001   
3       20      3        0      0          South Region      114555744   
4       20      4        0      0           West Region       71945553   
5       40      3        6      1               Alabama        4779736   
6       40      4        9      2                Alaska         710231   
7       40      4        8      4               Arizona        6392017   
8       40      3        7      5              Arkansas        2915918   
9       40      4        9      6            California       37253956   
10      40      4        8      8              Colorado        5029196   
11      40      1        1      9           Connecticut        3574097   
12      40      3        5     10              Delaware         897934   
13      40      3        5     11  District of Columbia         601723   
14      40      3        5     12               Florida       18801310   
15      40      3        5     13               Georgia        9687653   
16      40      4        9     15                Hawaii        1360301   
17      40      4        8     16                 Idaho        1567582   
18      40      2        3     17              Illinois       12830632   
19      40      2        3     18               Indiana        6483802   
20      40      2        4     19                  Iowa        3046355   
21      40      2        4     20                Kansas        2853118   
22      40      3        6     21              Kentucky        4339367   
23      40      3        7     22             Louisiana        4533372   
24      40      1        1     23                 Maine        1328361   
25      40      3        5     24              Maryland        5773552   
26      40      1        1     25         Massachusetts        6547629   
27      40      2        3     26              Michigan        9883640   
28      40      2        4     27             Minnesota        5303925   
29      40      3        6     28           Mississippi        2967297   
30      40      2        4     29              Missouri        5988927   
31      40      4        8     30               Montana         989415   
32      40      2        4     31              Nebraska        1826341   
33      40      4        8     32                Nevada        2700551   
34      40      1        1     33         New Hampshire        1316470   
35      40      1        2     34            New Jersey        8791894   
36      40      4        8     35            New Mexico        2059179   
37      40      1        2     36              New York       19378102   
38      40      3        5     37        North Carolina        9535483   
39      40      2        4     38          North Dakota         672591   
40      40      2        3     39                  Ohio       11536504   
41      40      3        7     40              Oklahoma        3751351   
42      40      4        9     41                Oregon        3831074   
43      40      1        2     42          Pennsylvania       12702379   
44      40      1        1     44          Rhode Island        1052567   
45      40      3        5     45        South Carolina        4625364   
46      40      2        4     46          South Dakota         814180   
47      40      3        6     47             Tennessee        6346105   
48      40      3        7     48                 Texas       25145561   
49      40      4        8     49                  Utah        2763885   
50      40      1        1     50               Vermont         625741   
51      40      3        5     51              Virginia        8001024   
52      40      4        9     53            Washington        6724540   
53      

In [18]:
# Son 52 örnek ve ilk 51 örneğin bulunduğu dataframe kısmında toplama yapalım
# Burada slicing işlem yapıyoruz aslında
state_pops.tail(52).head(51).sum(numeric_only=True)

sumlev                    2040
state                     1477
census2010pop        308745538
estimatesbase2010    308758105
popestimate2010      309327143
popestimate2011      311583481
popestimate2012      313877662
popestimate2013      316059947
popestimate2014      318386329
popestimate2015      320738994
popestimate2016      323071755
popestimate2017      325122128
popestimate2018      326838199
popestimate2019      328329953
popestimate042020    329398742
popestimate2020      329484123
dtype: int64

In [19]:
# count() metodu, dataframe'e ait olan sütunlarda toplam kaç adet veri bulunduğunu öğrenmemizi sağlar.
titanic.count()

pclass       1309
survived     1309
name         1309
sex          1309
age          1309
sibsp        1309
parch        1309
ticket       1309
fare         1309
cabin        1309
embarked     1309
boat         1309
body         1309
home.dest    1309
dtype: int64

In [20]:
# Netflix veri setini import ediyoruz
netflix = pd.read_csv("../data/netflix_titles.csv", sep="|", index_col=0)

In [21]:
netflix.head()

show_id     type                  title         director  \
0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1      s2  TV Show          Blood & Water              NaN   
2      s3  TV Show              Ganglands  Julien Leclercq   
3      s4  TV Show  Jailbirds New Orleans              NaN   
4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA   1 Season   
3  September 24, 2021          2021  TV-MA   1 Season   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

In [22]:
# Netflix dataframe'inin boyutlarını görelim
netflix.shape

(8807, 12)

In [23]:
# Netflix veri setinde bazı missing data'lar olup olmadığını görelim.
netflix.count()

show_id         8807
type            8807
title           8807
director        6173
cast            7982
country         7976
date_added      8797
release_year    8807
rating          8803
duration        8804
listed_in       8807
description     8807
dtype: int64

In [24]:
# mean() metodu, dataframe içerisindeki sütunlara ait değerlerin ortalamasını bulmamızı sağlar
houses.mean()

/home/can/anaconda3/envs/machineLearning/lib/python3.7/site-packages/ipykernel_launcher.py:2: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  


id               4.580302e+09
price            5.400881e+05
bedrooms         3.370842e+00
bathrooms        2.114757e+00
sqft_living      2.079900e+03
sqft_lot         1.510697e+04
floors           1.494309e+00
waterfront       7.541757e-03
view             2.343034e-01
condition        3.409430e+00
grade            7.656873e+00
sqft_above       1.788391e+03
sqft_basement    2.915090e+02
yr_built         1.971005e+03
yr_renovated     8.440226e+01
zipcode          9.807794e+04
lat              4.756005e+01
long            -1.222139e+02
sqft_living15    1.986552e+03
sqft_lot15       1.276846e+04
dtype: float64

In [25]:
# Yukarıda da görüldüğü üzere bir uyarı aldık. Buna göre mean() metodunu kullanırken non-numeric veriler ile çalışmak sakıncalı bir durum.
# O halde non-numeric verilerin ortalamasını almayacak şekilde mean() metodunu kullanalım.
houses.mean(numeric_only=True)

id               4.580302e+09
price            5.400881e+05
bedrooms         3.370842e+00
bathrooms        2.114757e+00
sqft_living      2.079900e+03
sqft_lot         1.510697e+04
floors           1.494309e+00
waterfront       7.541757e-03
view             2.343034e-01
condition        3.409430e+00
grade            7.656873e+00
sqft_above       1.788391e+03
sqft_basement    2.915090e+02
yr_built         1.971005e+03
yr_renovated     8.440226e+01
zipcode          9.807794e+04
lat              4.756005e+01
long            -1.222139e+02
sqft_living15    1.986552e+03
sqft_lot15       1.276846e+04
dtype: float64

In [26]:
titanic.mean(numeric_only=True)

pclass      2.294882
survived    0.381971
sibsp       0.498854
parch       0.385027
dtype: float64

In [27]:
# median() metodu, küçükten büyüğe sıralanmış bir dizi verinin ortasındaki değeri bulmamızı sağlar
houses.median(numeric_only=True)

id               3.904930e+09
price            4.500000e+05
bedrooms         3.000000e+00
bathrooms        2.250000e+00
sqft_living      1.910000e+03
sqft_lot         7.618000e+03
floors           1.500000e+00
waterfront       0.000000e+00
view             0.000000e+00
condition        3.000000e+00
grade            7.000000e+00
sqft_above       1.560000e+03
sqft_basement    0.000000e+00
yr_built         1.975000e+03
yr_renovated     0.000000e+00
zipcode          9.806500e+04
lat              4.757180e+01
long            -1.222300e+02
sqft_living15    1.840000e+03
sqft_lot15       7.620000e+03
dtype: float64

In [28]:
titanic.median(numeric_only=True)

pclass      3.0
survived    0.0
sibsp       0.0
parch       0.0
dtype: float64

In [29]:
# mode() metodu, data rame'e ait sütunlardaki verilerden en çok tekrar edenleri öğrenmemizi sağlayar
titanic.mode(numeric_only=True)

pclass  survived  sibsp  parch
0       3         0      0      0

In [30]:
houses.mode(numeric_only=True)

id     price  bedrooms  bathrooms  sqft_living  sqft_lot  floors  \
0  795000620.0  350000.0       3.0        2.5       1300.0    5000.0     1.0   
1          NaN  450000.0       NaN        NaN          NaN       NaN     NaN   
2          NaN       NaN       NaN        NaN          NaN       NaN     NaN   
3          NaN       NaN       NaN        NaN          NaN       NaN     NaN   

   waterfront  view  condition  grade  sqft_above  sqft_basement  yr_built  \
0         0.0   0.0        3.0    7.0      1300.0            0.0    2014.0   
1         NaN   NaN        NaN    NaN         NaN            NaN       NaN   
2         NaN   NaN        NaN    NaN         NaN            NaN       NaN   
3         NaN   NaN        NaN    NaN         NaN            NaN       NaN   

   yr_renovated  zipcode      lat    long  sqft_living15  sqft_lot15  
0           0.0  98103.0  47.5322 -122.29         1540.0      5000.0  
1           NaN      NaN  47.5491     NaN            NaN         NaN  
2           NaN      NaN  47.6624     NaN            NaN         NaN  
3           NaN      NaN  47.6846     NaN            NaN         NaN

In [31]:
# describe(), metodu, Yalnızca sayısal verilere ait bazı istatistiksel bilgileri bize sunar.
titanic.describe()

pclass     survived        sibsp        parch
count  1309.000000  1309.000000  1309.000000  1309.000000
mean      2.294882     0.381971     0.498854     0.385027
std       0.837836     0.486055     1.041658     0.865560
min       1.000000     0.000000     0.000000     0.000000
25%       2.000000     0.000000     0.000000     0.000000
50%       3.000000     0.000000     0.000000     0.000000
75%       3.000000     1.000000     1.000000     0.000000
max       3.000000     1.000000     8.000000     9.000000

In [32]:
houses.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [33]:
type(houses.describe())

pandas.core.frame.DataFrame

***
**`describe()` metodunu nümerik verilerle kullandık. Peki nümerik olmayan veriler ile kullanabilir miyiz?**
***

In [34]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1309 entries, 0 to 1308
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   pclass     1309 non-null   int64 
 1   survived   1309 non-null   int64 
 2   name       1309 non-null   object
 3   sex        1309 non-null   object
 4   age        1309 non-null   object
 5   sibsp      1309 non-null   int64 
 6   parch      1309 non-null   int64 
 7   ticket     1309 non-null   object
 8   fare       1309 non-null   object
 9   cabin      1309 non-null   object
 10  embarked   1309 non-null   object
 11  boat       1309 non-null   object
 12  body       1309 non-null   object
 13  home.dest  1309 non-null   object
dtypes: int64(4), object(10)
memory usage: 143.3+ KB


In [35]:
# Yalnızca object tipindeki sütunlar üzerinde describe() metodunun çalıştırılması.
# Bu işlemi "include" parametresi sayesinde gerçekleştirebiliyoruz.
titanic.describe(include=["object"])

name   sex   age    ticket  fare cabin embarked  boat  \
count                   1309  1309  1309      1309  1309  1309     1309  1309   
unique                  1307     2    99       929   282   187        4    28   
top     Connolly, Miss. Kate  male     ?  CA. 2343  8.05     ?        S     ?   
freq                       2   843   263        11    60  1014      914   823   

        body home.dest  
count   1309      1309  
unique   122       370  
top        ?         ?  
freq    1188       564

In [36]:
# Yalnızca object tipindeki sütunlar üzerinde describe() metodunun çalıştırılması.
titanic.describe(include=["O"])

name   sex   age    ticket  fare cabin embarked  boat  \
count                   1309  1309  1309      1309  1309  1309     1309  1309   
unique                  1307     2    99       929   282   187        4    28   
top     Connolly, Miss. Kate  male     ?  CA. 2343  8.05     ?        S     ?   
freq                       2   843   263        11    60  1014      914   823   

        body home.dest  
count   1309      1309  
unique   122       370  
top        ?         ?  
freq    1188       564

In [37]:
# Birbirinden farklı veri tiplerine sahip sütunlar için describe() metodunun çalıştırılması
titanic.describe(include=["object", "int64"])

pclass     survived                  name   sex   age  \
count   1309.000000  1309.000000                  1309  1309  1309   
unique          NaN          NaN                  1307     2    99   
top             NaN          NaN  Connolly, Miss. Kate  male     ?   
freq            NaN          NaN                     2   843   263   
mean       2.294882     0.381971                   NaN   NaN   NaN   
std        0.837836     0.486055                   NaN   NaN   NaN   
min        1.000000     0.000000                   NaN   NaN   NaN   
25%        2.000000     0.000000                   NaN   NaN   NaN   
50%        3.000000     0.000000                   NaN   NaN   NaN   
75%        3.000000     1.000000                   NaN   NaN   NaN   
max        3.000000     1.000000                   NaN   NaN   NaN   

              sibsp        parch    ticket  fare cabin embarked  boat  body  \
count   1309.000000  1309.000000      1309  1309  1309     1309  1309  1309   
unique          NaN          NaN       929   282   187        4    28   122   
top             NaN          NaN  CA. 2343  8.05     ?        S     ?     ?   
freq            NaN          NaN        11    60  1014      914   823  1188   
mean       0.498854     0.385027       NaN   NaN   NaN      NaN   NaN   NaN   
std        1.041658     0.865560       NaN   NaN   NaN      NaN   NaN   NaN   
min        0.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
25%        0.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
50%        0.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
75%        1.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
max        8.000000     9.000000       NaN   NaN   NaN      NaN   NaN   NaN   

       home.dest  
count       1309  
unique       370  
top            ?  
freq         564  
mean         NaN  
std          NaN  
min          NaN  
25%          NaN  
50%          NaN  
75%          NaN  
max          NaN

In [38]:
# Dataframe içerisinde bulunan bütün veri tipleri için describe() metodunun çalıştırılması
titanic.describe(include="all")

pclass     survived                  name   sex   age  \
count   1309.000000  1309.000000                  1309  1309  1309   
unique          NaN          NaN                  1307     2    99   
top             NaN          NaN  Connolly, Miss. Kate  male     ?   
freq            NaN          NaN                     2   843   263   
mean       2.294882     0.381971                   NaN   NaN   NaN   
std        0.837836     0.486055                   NaN   NaN   NaN   
min        1.000000     0.000000                   NaN   NaN   NaN   
25%        2.000000     0.000000                   NaN   NaN   NaN   
50%        3.000000     0.000000                   NaN   NaN   NaN   
75%        3.000000     1.000000                   NaN   NaN   NaN   
max        3.000000     1.000000                   NaN   NaN   NaN   

              sibsp        parch    ticket  fare cabin embarked  boat  body  \
count   1309.000000  1309.000000      1309  1309  1309     1309  1309  1309   
unique          NaN          NaN       929   282   187        4    28   122   
top             NaN          NaN  CA. 2343  8.05     ?        S     ?     ?   
freq            NaN          NaN        11    60  1014      914   823  1188   
mean       0.498854     0.385027       NaN   NaN   NaN      NaN   NaN   NaN   
std        1.041658     0.865560       NaN   NaN   NaN      NaN   NaN   NaN   
min        0.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
25%        0.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
50%        0.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
75%        1.000000     0.000000       NaN   NaN   NaN      NaN   NaN   NaN   
max        8.000000     9.000000       NaN   NaN   NaN      NaN   NaN   NaN   

       home.dest  
count       1309  
unique       370  
top            ?  
freq         564  
mean         NaN  
std          NaN  
min          NaN  
25%          NaN  
50%          NaN  
75%          NaN  
max          NaN

In [39]:
netflix.describe(include="O")

show_id   type                 title       director  \
count     8807   8807                  8807           6173   
unique    8807      2                  8807           4528   
top         s1  Movie  Dick Johnson Is Dead  Rajiv Chilaka   
freq         1   6131                     1             19   

                      cast        country       date_added rating  duration  \
count                 7982           7976             8797   8803      8804   
unique                7692            748             1767     17       220   
top     David Attenborough  United States  January 1, 2020  TV-MA  1 Season   
freq                    19           2818              109   3207      1793   

                           listed_in  \
count                           8807   
unique                           514   
top     Dramas, International Movies   
freq                             362   

                                              description  
count                                                8807  
unique                                               8775  
top     Paranormal activity at a lush, abandoned prope...  
freq                                                    4